### PR TITLE
Add trigger dag option to deployment workflow

### PR DESCRIPTION
**Pull request by Lee-W**
## Why
Most of the time, I use the [deploy-integration-tests-to-astro-cloud](https://github.com/astronomer/astronomer-providers/actions/workflows/deploy-integration-tests.yaml) for testing certain DAGs. Thus, if we can add an option to let the workflow which DAGs should be triggered after the deployment, it would facilitate our testing process

## Changes
* github-actions
    * add dags_to_trigger_after_deployment input to deployment workflow
* dev/providers_rc_test_scripts
    * change trigger_master_dag into trigger_dag which can take argument dag_ids to trigger multiple different dags

### CODE REVIEW

**Comment by Lee-W** [comment]
image

**Comment by Lee-W** [comment]
Example run
https://github.com/astronomer/astronomer-providers/actions/runs/5068141893

File: .github/workflows/deploy-integration-tests.yaml [hunk 1]
```diff
@@ -55,10 +64,11 @@ jobs:
 
       - name: Trigger master dag
```
Comment (pankajkoti) [inline .github/workflows/deploy-integration-tests.yaml:65]:
```suggestion
      - name: Trigger DAG(s)
```

File: .github/workflows/deploy-integration-tests.yaml [hunk 2]
```diff
@@ -37,7 +44,9 @@ jobs:
       SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
 
   wait-for-deployment-to-be-ready-and-trigger-master-dag-for-providers-integration-tests:
```
Comment (pankajkoti) [inline .github/workflows/deploy-integration-tests.yaml:46]:
```suggestion
  wait-for-deployment-to-be-ready-and-trigger-dag(s)-for-providers-integration-tests:
```

File: .github/workflows/deploy-integration-tests.yaml [hunk 3]
```diff
@@ -77,7 +87,9 @@ jobs:
       SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
 
   wait-for-deployment-to-be-ready-and-trigger-master-dag-for-providers-integration-tests-on-KE:
```
Comment (pankajkoti) [inline .github/workflows/deploy-integration-tests.yaml:89]:
```suggestion
  wait-for-deployment-to-be-ready-and-trigger-dag(s)-for-providers-integration-tests-on-KE:
```

File: .github/workflows/deploy-integration-tests.yaml [hunk 4]
```diff
@@ -104,7 +116,8 @@ jobs:
 
       - name: Trigger master dag
```
Comment (pankajkoti) [inline .github/workflows/deploy-integration-tests.yaml:117]:
```suggestion
      - name: Trigger dag(s)
```

File: .github/workflows/test-rc-release.yaml [hunk 5]
```diff
@@ -141,7 +141,7 @@ jobs:
 
       - name: Trigger master dag
         run: |
-          python3 dev/providers_rc_test_scripts/trigger_master_dag.py \
+          python3 dev/integration_test_scripts/trigger_master_dag.py \
```
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:144]:
```suggestion
          python3 dev/integration_test_scripts/trigger_dag.py \
```

File: dev/integration_test_scripts/trigger_dag.py [hunk 6]
```diff
@@ -28,31 +27,48 @@ def get_access_token(api_key_id: str, api_key_secret: str) -> str:
...
     parser.add_argument("deployment_id", help="ID of the deployment in Astro Cloud")
     parser.add_argument("astronomer_key_id", help="Key ID of the Astro Cloud deployment")
     parser.add_argument("astronomer_key_secret", help="Key secret of the Astro Cloud deployment")
+    parser.add_argument(
+        "--dag-ids",
+        help=(
+            "Comma separated list of dag_ids to trigger"
+            " e.g. 'example_async_adf_run_pipeline, example_async_batch'"
+        ),
+    )
```
Comment (pankajkoti) [inline dev/integration_test_scripts/trigger_dag.py:63]:
```suggestion
        ),
        default="example_master_dag",
    )
```

File: dev/integration_test_scripts/trigger_dag.py [hunk 7]
```diff
@@ -28,31 +27,48 @@ def get_access_token(api_key_id: str, api_key_secret: str) -> str:
...
+
     args = parser.parse_args()
     token = get_access_token(args.astronomer_key_id.strip(), args.astronomer_key_secret.strip())
-    trigger_master_dag_run(args.deployment_id, token)
+
+    input_dag_ids = args.dag_ids
+    if not input_dag_ids:
+        dag_ids = ["example_master_dag"]
+    else:
+        dag_ids = [dag_id.strip() for dag_id in input_dag_ids.split(",")]
```
Comment (pankajkoti) [inline dev/integration_test_scripts/trigger_dag.py:72]:
```suggestion
    dag_ids = [dag_id.strip() for dag_id in input_dag_ids.split(",")]
```
If we accept the above suggestion for having a default value, then we don't need this branches.

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**